pulsar/reader_impl.go

```diff
@@ -207,12 +191,15 @@ func (r *reader) Seek(msgID MessageID) error {
 		return nil
 	}
 
-	return r.pc.Seek(mid)
+	return r.consumers[mid.partitionIdx].Seek(mid)
```

Comment:
could there be an invalid partition (out of range) index here?

Reply:
why？The number of Consumers is determined by the TopicPartitions() function. If there is a problem here, then the Patition_Consumer is also problematic.

Reply:
@cckellogg